Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -26,120 +26,68 @@ public class NetworkWriter
     {
         public const int MaxStringLength = 1024 * 32;
 
-        // create writer immediately with it's own buffer so no one can mess with it and so that we can resize it.
-        // note: BinaryWriter allocates too much, so we only use a MemoryStream
-        // => 1500 bytes by default because on average, most packets will be <= MTU
-        byte[] buffer = new byte[1500];
+        // fixed size buffer so we can pool it and don't need runtime resizing.
+        readonly byte[] buffer;
 
         // 'int' is the best type for .Position. 'short' is too small if we send >32kb which would result in negative .Position
         // -> converting long to int is fine until 2GB of data (MAX_INT), so we don't have to worry about overflows here
-        int position;
-        int length;
+        public int Position;
 
-        public int Length => length;
+        // helper field to calculate space in bytes remaining to write
+        public int Space => buffer.Length - Position;
```

Comment:
Remember that properties like this have methods so have performance cost. We only use this in 2 places maybe we should just in-line it for now.

If we want to keep the property then maybe rename it to `Remaining` so that the comment isnt needed to explain what it does.